src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerAssemblyLoader.Core.cs

```diff
@@ -18,58 +18,60 @@
-    }
-
-    internal partial class AnalyzerAssemblyLoader
-    {
-        private readonly AssemblyLoadContext _compilerLoadContext;
         private readonly Dictionary<string, DirectoryLoadContext> _loadContextByDirectory = new Dictionary<string, DirectoryLoadContext>(StringComparer.Ordinal);
-        private readonly AnalyzerLoadOption _loadOption;
 
-        internal AssemblyLoadContext CompilerLoadContext => _compilerLoadContext;
-        internal AnalyzerLoadOption AnalyzerLoadOption => _loadOption;
+        public IAnalyzerAssemblyResolver CompilerAnalyzerAssemblyResolver { get; }
+        public AssemblyLoadContext CompilerLoadContext { get; }
+        public ImmutableArray<IAnalyzerAssemblyResolver> AnalyzerAssemblyResolvers { get; }
 
-        internal AnalyzerAssemblyLoader(ImmutableArray<IAnalyzerAssemblyResolver> externalResolvers)
-            : this(null, AnalyzerLoadOption.LoadFromDisk, externalResolvers)
+        internal AnalyzerAssemblyLoader()
+            : this([], [DiskAnalyzerAssemblyResolver], null)
         {
         }
 
-        internal AnalyzerAssemblyLoader(AssemblyLoadContext? compilerLoadContext, AnalyzerLoadOption loadOption, ImmutableArray<IAnalyzerAssemblyResolver> externalResolvers)
+        /// <summary>
+        /// Create a new <see cref="AnalyzerAssemblyLoader"/> with the given resolvers.
+        /// </summary>
+        /// <param name="pathResolvers"></param>
+        /// <param name="assemblyResolvers"></param>
+        /// <param name="compilerLoadContext">This is the <see cref="AssemblyLoadContext"/> where the compiler resides. This parameter
+        /// is primarily used for testing purposes but is also useful in hosted scenarios where the compiler may be loaded outside
+        /// the default context.</param>
+        /// <exception cref="ArgumentException"></exception>
+        internal AnalyzerAssemblyLoader(
+            ImmutableArray<IAnalyzerPathResolver> pathResolvers,
+            ImmutableArray<IAnalyzerAssemblyResolver> assemblyResolvers = default,
+            AssemblyLoadContext? compilerLoadContext = null)
         {
-            _loadOption = loadOption;
-            _compilerLoadContext = compilerLoadContext ?? AssemblyLoadContext.GetLoadContext(typeof(AnalyzerAssemblyLoader).GetTypeInfo().Assembly)!;
-            _externalResolvers = [.. externalResolvers, new CompilerAnalyzerAssemblyResolver(_compilerLoadContext)];
+            if (assemblyResolvers.Length == 0)
+            {
+                throw new ArgumentException("Cannot be default", nameof(assemblyResolvers));
+            }
+
+            CompilerLoadContext = compilerLoadContext ?? AssemblyLoadContext.GetLoadContext(typeof(SyntaxTree).GetTypeInfo().Assembly)!;
+            CompilerAnalyzerAssemblyResolver = new CompilerResolver(CompilerLoadContext);
+            AnalyzerPathResolvers = pathResolvers;
+            AnalyzerAssemblyResolvers = [CompilerAnalyzerAssemblyResolver, .. assemblyResolvers];
```

Comment:
is it important that CompilerAnalyzerAssemblyResolver be first in this list?  (If so,  or if not, a comment might be helpful stating as such).